sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py

```diff
@@ -314,3 +315,18 @@ def test_default_labels(pyplot, display_labels, expected_labels):
 
     assert_array_equal(x_ticks, expected_labels)
     assert_array_equal(y_ticks, expected_labels)
+
+
+def test_error_on_a_dataset_with_unseen_labels(pyplot, fitted_clf, data):
+    X, y = data
+
+    # get values that are not present in 'fitted_clf.classes_'
+    y = y + 1
+    disp = plot_confusion_matrix(fitted_clf, X, y,
+                                 labels=None, display_labels=None)
+
+    disp_labels = set([tick.get_text() for tick in disp.ax_.get_xticklabels()])
+    expected_labels = unique_labels(y, fitted_clf.predict(X))
```

Comment:
In this case, we can list the labels:

```py
    display_labels = [tick.get_text() for tick in disp.ax_.get_xticklabels()]
    expected_labels = [f'{i}' for range(6)]
    assert_array_equal(expected_labels, display_labels)
```

Reply:
Thank you, I have replaced these lines and the assertion check with your code.